test/metabase/query_processor/metadata_test.clj

```diff
@@ -0,0 +1,101 @@
+(ns metabase.query-processor.metadata-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.driver :as driver]
+   [metabase.query-processor.metadata :as qp.metadata]
+   [metabase.test :as mt]))
+
+(deftest ^:parallel mbql-query-metadata-test
+  (testing "Should be able to calculate metadata for an MBQL query without going in to driver land"
+    (is (=? [{:lib/type      :metadata/column
+              :name          "ID"
+              :database-type "BIGINT"
+              :base-type     :type/BigInteger
+              :id            (mt/id :venues :id)}
+             {:lib/type      :metadata/column
+              :name          "NAME"
+              :database-type "CHARACTER VARYING"
+              :base-type     :type/Text
+              :id            (mt/id :venues :name)}
+             {:lib/type      :metadata/column
+              :name          "CATEGORY_ID"
+              :database-type "INTEGER"
+              :base-type     :type/Integer
+              :id            (mt/id :venues :category_id)}
+             {:lib/type      :metadata/column
+              :name          "LATITUDE"
+              :database-type "DOUBLE PRECISION"
+              :base-type     :type/Float
+              :id            (mt/id :venues :latitude)}
+             {:lib/type      :metadata/column
+              :name          "LONGITUDE"
+              :database-type "DOUBLE PRECISION"
+              :base-type     :type/Float
+              :id            (mt/id :venues :longitude)}
+             {:lib/type      :metadata/column
+              :name          "PRICE"
+              :database-type "INTEGER"
+              :base-type     :type/Integer
+              :id            (mt/id :venues :price)}]
+            (qp.metadata/result-metadata (mt/mbql-query venues))))))
+
+(deftest ^:parallel native-query-metadata-test
```

Comment:
For other drivers it gets tested indirectly in a bunch of other tests, I didn't want to overcomplicate this test since the output is going to differ completely between all of the different drivers